crates/toolchain/instructions/src/program.rs

```diff
@@ -68,11 +75,11 @@ impl<F: Field> Program<F> {
         }
     }
 
-    pub fn add_apc_instruction_at_pc_index(&mut self, pc_index: usize, opcode: VmOpcode) {
+    pub fn add_apc_instruction_at_pc_index(&mut self, pc_index: usize, opcode: VmOpcode, condition: ApcCondition) {
```

Comment:
This is the user facing change, where the client specifies the conditions under which the apc should be chosen